<launch>

    <arg name="epuck_address"/>
    <arg name="epuck_name" default="epuck_robot_0"/>
    <arg name="cam_en" default="false"/> <!-- camera -->
    <arg name="acc_en" default="true"/> <!-- acelerometro -->
    <arg name="prox_en" default="true"/> <!-- proximidade -->
    <arg name="mot_pos_en" default="true"/> <!-- posição angular da roda -->
    <arg name="light_en" default="true"/> <!-- sensor de luminozidade superior -->
    <arg name="floor_en" default="false"/> <!-- sensor de luminozidade inferior -->
	<arg name="sel_en" default="true"/> <!-- botao seletor -->
	<arg name="mot_speed_en" default="true"/> <!-- velocidade angular do motor -->
	<arg name="mic_en" default="false"/> <!-- microfones -->
	<arg name="xpos" default="0.0"/> <!-- odometria -->
	<arg name="ypos" default="0.0"/> <!-- odometria -->
	<arg name="theta" default="0.0"/> <!-- odometria -->
	<arg name="is_single_robot" default="1" /> <!-- é robo unico -->

    <node pkg="epuck_driver" type="epuck_driver.py" name="$(arg epuck_name)" output="screen">
        <param name="epuck_address" value="$(arg epuck_address)"/>
        <param name="epuck_name" value="$(arg epuck_name)"/>
        <param name="camera" value="$(arg cam_en)"/>
        <param name="accelerometer" value="$(arg acc_en)"/>
        <param name="proximity" value="$(arg prox_en)"/>
        <param name="motor_position" value="$(arg mot_pos_en)"/>
        <param name="light" value="$(arg light_en)"/>
        <param name="floor" value="$(arg floor_en)"/>
		<param name="selector" value="$(arg sel_en)"/>
		<param name="motor_speed" value="$(arg mot_speed_en)"/>
		<param name="microphone" value="$(arg mic_en)"/>
		<param name="xpos" value="$(arg xpos)"/>
		<param name="ypos" value="$(arg ypos)"/>
		<param name="theta" value="$(arg theta)"/>
    </node>

    <param name="robot_description" textfile="$(find epuck_driver)/urdf/epuck_urdf.xml"/>

    <node name="epuck_state_publisher" pkg="robot_state_publisher" type="state_publisher">
        <param name="tf_prefix" value="$(arg epuck_name)" />
    </node>

    <!--node if="$(arg is_single_robot)" pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find epuck_driver)/config/single_epuck_driver_rviz.rviz"/-->

</launch>
